tt_metal/jit_build/CMakeLists.txt

```
@@ -8,3 +8,7 @@ set(JIT_BUILD_SRCS
 
 add_library(jit_build OBJECT ${JIT_BUILD_SRCS})
 target_link_libraries(jit_build PUBLIC common)
+
+if(DEFINED VERSION_HASH)
+    target_compile_definitions(jit_build PUBLIC "-DGIT_COMMIT_HASH=\"${VERSION_HASH}\"")
```

`PUBLIC`?  This will have serious implications on builds.  It'll force a new define on all consumers.  I don't see anyone outside of `jit_build` using it, so that's just unnecessary make-work for the compiler.

```suggestion
    target_compile_definitions(jit_build PRIVATE GIT_COMMIT_HASH=${VERSION_HASH})
```